model/criteria/operators.go

```diff
@@ -227,3 +228,46 @@ func inPeriod(m map[string]interface{}, negate bool) (Expression, error) {
 func startOfPeriod(numDays int64, from time.Time) string {
 	return from.Add(time.Duration(-24*numDays) * time.Hour).Format("2006-01-02")
 }
+
+type InPlaylist map[string]interface{}
+
+func (ipl InPlaylist) ToSql() (sql string, args []interface{}, err error) {
+	return inList(ipl, false)
+}
+
+func (ipl InPlaylist) MarshalJSON() ([]byte, error) {
+	return marshalExpression("inPlaylist", ipl)
+}
+
+type NotInPlaylist map[string]interface{}
+
+func (ipl NotInPlaylist) ToSql() (sql string, args []interface{}, err error) {
+	return inList(ipl, true)
+}
+
+func (ipl NotInPlaylist) MarshalJSON() ([]byte, error) {
+	return marshalExpression("notInPlaylist", ipl)
+}
+
+func inList(m map[string]interface{}, negate bool) (sql string, args []interface{}, err error) {
+	var playlistname string
+	var ok bool
+	if playlistname, ok = m["name"].(string); !ok {
+		return "", nil, errors.New("playlist name not given")
+	}
+
+	// Subquery to check if media_file.id matches with the playlist name in playlist_tracks table
+	subQuery := squirrel.Select("1").
+		From("playlist_tracks pl").
+		LeftJoin("playlist on pl.playlist_id = playlist.id").
+		Where(squirrel.And{squirrel.Expr("pl.media_file_id = media_file.id"), squirrel.Eq{"playlist.name": playlistname}})
```

Comment:
Also maybe we need to check the owner id of the playlist?

On the one hand, a user shouldn't be able to hijack tracks into another user's smart playlist just by giving their playlist a conflicting name. On the other hand, a user should also not be able to snoop on other users' playlists by creating a smart playlist.

Reply:
Do note that there is no constraint in the names being unique, so you should definitely check for the user ownership. 
It is possible for one user to have multiple playlists of the same name as well, so you should give the option of providing the ID instead

Reply:
In the current interface of adding smart playlist, there is no way that user can specify the id of a playlist, but on the other hand, in the future when we have UI to add smart playlist, it's unlikely we want to support using name matching. So maybe in this case we want to translate name to id when importing?

But the same issue probably applies for all other criteria, so maybe it's not a big issue.

Reply:
That first point isn't entirely true. You can definitely reference the ID of an existing playlist, you would just need to add that SQL in the rule, so you could have something like
```json
{
    "all":  [
        {"inPlaylist": {"id": "guid" }}
    ],  
    "sort": "random"
}
```
Of course, this could run into the problem if you're referencing other playlists, and suppose you migrate to a new system but /shrug/. You could still support both.
Also I guess it is worth point out that smart playlists (currently) are created for the first admin user, so idk how permissions would really shake out.

Reply:
But how can a user be expected to know what the playlist id is? I don't think everyone knows how to do SQL 😄

Reply:
You don't need SQL for that, it's part of the page URL (for example):
`http://localhost:4533/#/playlist/7b425f26-4659-4d4b-9a8c-9321866e8a2c/show` ID: "7b425f26-4659-4d4b-9a8c-9321866e8a2c"

Reply:
Oh okay, that's easier then.

Reply:
In that case I think we should only support id.

Reply:
There is one benefit I can think of for also supporting names:

Suppose you are creating multiple playlists at once, and at least one of them is a smart playlist. 
You know what all their names are going to be (by the file name), but the ID is unknown until creation. 
So, suppose you had files A, B, C, and D depending on A, and E depending on D and B (this is a contrived example).
If you support by name, Navidrome will _eventually_ figure out the relationships and all playlists will be filled.
By contrast, if you need an ID, then you couldn't create D or E until A, and C are created.

Is that worth it? I don't know. But it is a use case. Regardless, I definitely thing ID should be a possible field

Reply:
As I mentioned before, when we have UI to create playlist, I think it's more likely we would just let user pick a playlist for existing ones, so unlikely name would be useful.

Reply:
Regarding the playlist owner, I'd love to be able to combine playlists of different users.

A use case would be that person A, B and C create their own playlist "roadtrip". And when the three go on a roadtrip, they create a smart playlist to merge the three playlists dynamically into one.

Reply:
The playlist should at least be shared I guess.

But if we use id rather than name, it is less of a concern to accidentally include or leak tracks of a playlist.